flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/SubtaskGatewayImpl.java

```diff
@@ -207,8 +204,8 @@ void markForCheckpoint(long checkpointId) {
     boolean tryCloseGateway(long checkpointId) {
         checkRunsInMainThread();
 
-        if (checkpointId == currentCheckpointId) {
-            isClosed = true;
+        if (currentMarkedCheckpointIds.contains(checkpointId)) {
+            blockedEventsMap.putIfAbsent(checkpointId, new LinkedList<>());
```

Comment:
Would it be more readable to do `put()` and throw exception if `blockedEventsMap()` already contains the given `checkpointId`?